src/compute-client/src/command.rs

```diff
@@ -1071,13 +1073,26 @@ impl<T: timely::progress::Timestamp> ComputeCommandHistory<T> {
         command: ComputeCommand<T>,
         peeks: &std::collections::HashMap<uuid::Uuid, V>,
     ) {
+        if let ComputeCommand::CreateDataflows(dataflows) = &command {
+            self.dataflow_count += dataflows.len();
```

Comment:
Don't we need to decrement this when a dataflow is dropped?

Reply:
At least for now, my intention was to track the number of dataflows in the current version of the history (live or not), so we only need to make sure to reflect a decrease when the history is reduced. I think that the metric you have in mind, the number of live dataflows in the history, would also be very helpful, but I would suggest it be delayed to another PR. The number of dataflows (live or not) in the current version of the history lets us see if we are leaving behind dataflow descriptions once slow-path peeks are retired and we reduce the history to get rid of them.

Reply:
PS: There is also some overlap between tracking live dataflows and tracking the number of collections that we keep in the compute controller. So we might be happy enough with the tracking the latter as a metric, which would support #16028.